src/Config/ObjectModel/RelationshipRole.cs

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Azure.DataApiBuilder.Config.ObjectModel;
+
+/// <summary>
+/// Denotes the role of a referencing or referenced table in a relationship.
+/// Default value is None because the role is only resolved in some scenarios
+/// such as self-referencing relationships.
+/// </summary>
+public enum RelationshipRole
+{
+    None,
+    Source,
```

Comment:
clarified comment:
```csharp
/// Denotes the role of a referencing or referenced table in a relationship.
/// Default value is None because the role is only resolved to help DAB
/// create join predicates for self-referencing relationships.
/// In non self-join relationships, DAB uses the ForeignKeyDefinition.RelationShipPair
/// to determine the referencing/referenced entity. RelationShipPair isn't sufficient
/// for self-join relationships because it the DatabaseObjects used to represent the pair
/// reference the same object: e.g. both Referenced/Referencing entity would point to 'dbo.MyTable'
```

Reply:
Would be useful to identify what a source means? what a target means? IIUC, its with respect to how the configuration is layed out.

Reply:
You're right, Ani. The concept of 'source' originates from the DAB config. 
- **Source** -> entity which defines a relationship between it (the source) and a target entity (defined by "targetentity" property).

In GraphQL query, source is the parent entity and target is the sub-entity.
![image](https://github.com/Azure/data-api-builder/assets/6414189/738eb71d-49e6-4945-a1a5-fdd60103ad49)